### ISSUE TITLE
[v4.0.5] Problem returning the outDir

### ISSUE THREAD

**Issue by bogdancondorachi** [arnoson/kirby-vite#27]
I've tried to update the plugin and my setup to the v4 but keep getting:

`Cannot assign bool to property arnoson\KirbyVite\Vite::$outDir of type ?string`

That happens on the outDir function:

```php
protected function outDir() {
    return $this->outDir ??= kirby()->root('base')
      ? getRelativePath(
        kirby()->root('index'),
        kirby()->root('base') . '/' . $this->config()['outDir']
      )
      : $this->config()['outDir'];
  }
``` 
  
I've also tried both starter kit's and got the same results without luck on getting to the bottom of it.

**EDIT:**
I've got through the above error by adding `$this->outDir ??= kirby()->root('base')` inside parentheses to check whether $this->outDir is null and then decide whether to calculate the output directory path based on the condition
```php
return ($this->outDir ??= kirby()->root('base'))
``` 

But forward it seems like the manifest path is wrong, returning `public//manifest.json` meaning that the $outDir returns empty in the following case
```php
$manifestPath = "$index/$outDir/manifest.json";
``` 

_*The tests are made on the vite starter kit with just the modifications presented above._

**Comment by arnoson** [comment]
Are you using a public folder setup? I just checked and for me both the basic and the multi page kit are working. I'm running it on WSL 2 (Ubuntu). Do I anderstand that plain kits are not working for you? Or have you modified the vite.config.js somehow?

**Comment by arnoson** [comment]
Okay, I'm not coding in Windows so I don't have composer/php ready to test it. Some thoughts:
- If you want to fake the outDir you would have tu return `dist` instead of `kirby()->root('base')`
- Is the `vite.config.php` generated correctly inside your `site/config` folder? It should contain
```php
<?php return [
  'outDir' => 'public/dist',
  'assetsDir' => 'assets',
  'legacy' => false
];
```
- I'm taking vacations from tomorrow on so I wont be able to look into this issue for a while. So either try it in Linux/WSL or stick with version 3.x until I can test it Windows. Might be a file path issue (maybe caused  by `getRelativePath` using `/`)

**Comment by bogdancondorachi** [comment]
I've looked more into the getRelativePath function and indeed it's related to file path differences between the operating systems.

I've added `realpath()` to standardize and resolve paths and `DIRECTORY_SEPARATOR` to represent the correct directory separator for the current operating system.

```php
function getRelativePath($rootPath, $fullPath) {
  $rootPath = realpath(rtrim($rootPath, '/'));
  $fullPath = realpath(rtrim($fullPath, '/'));

  if (strpos($fullPath, $rootPath) === 0) {
    $relativePath = ltrim(substr($fullPath, strlen($rootPath)), DIRECTORY_SEPARATOR);
    return $relativePath;
  }

  return false;
}
``` 

Let me know your thoughts on this and if you can test it on linux as well. Also do you want to open an PR for this?

**Comment by arnoson** [comment]
I could reproduce your issue in windows now and your `getRelativePath` works great in both windows and linux. Thanks for pointing out this issue and coming up with a fix! Do you want to create a PR with these changes? I could merge it later today

**Comment by bogdancondorachi** [comment]
Great! Thanks as well for pointing me in the right direction. I've added the PR #28

**Comment by arnoson** [comment]
Released in `v4.0.6` :)
